### PR TITLE
[CIVP-11405] BLD: Fix functools32 installation

### PR DESCRIPTION
**Pull request by jseabold**
Hopefully fixes the following error on Python 3, but I haven't been able to replicate not installing from PyPI.

```
Using cached functools32-3.2.3-2.zip
Complete output from command python setup.py egg_info:
This backport is for Python 2.7 only.
```

Issue is described in the following

https://github.com/inveniosoftware/invenio-oaiserver/issues/112
https://github.com/Julian/jsonschema/issues/228
https://hynek.me/articles/conditional-python-dependencies/
https://wheel.readthedocs.io/en/latest/#defining-conditional-dependencies

And the differing usage of install_requires vs requirements.txt here

https://packaging.python.org/requirements/

### CODE REVIEW

**Comment by keithing** [comment]
@jseabold we need to make a patch release for this bugfix.  Do you mind merging against `release-1.5` instead of `master`?

**Comment by keithing** [comment]
Do you mind also adding a line to the changelog?

**Comment by keithing** [comment]
@jseabold, actually hold on for a second before changing the merge branch.  We're settling on the naming convention for release branches.

**Comment by keithing** [comment]
Okay, go ahead and switch the merge into branch to `release-1.5`.  (We originally were using `release-v1.5`, but are going to move ahead without the leading v for versions).

**Comment by peterskipper** [comment]
Has this been merged? I'm running into this problem trying to reconstruct an environment with conda

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 pyyaml>=3.0,<=3.99
 click>=6.0,<=6.99
 funcsigs==1.0.2 ; python_version == '2.7'
-future>-0.16,<=0.99 ; python_version == '2.7'
+future>=0.16,<=0.99 ; python_version == '2.7'
```
Comment (keithing) [inline requirements.txt]: If we're no longer basing our requirements off `requirements.txt`, it might be a good idea to remove it (to avoid confusion).

Comment (stephen-hoover) [inline requirements.txt]: Or maybe parse the requirements file to fill the `install_requires` and `extras_require` fields.

Comment (jseabold) [inline requirements.txt]: I specifically did not want to parse here. They have different uses as linked in the PR. If someone wants to pin requirements.txt to versions for its intended use. that's fine, but probably a different PR.

Comment (keithing) [inline requirements.txt]: > They have different uses as linked in the PR. If someone wants to pin requirements.txt to versions for its intended use. that's fine, but probably a different PR

Yeah, let's not get into pinning versions here.

**Pull request merged**